Add links to text_expansion in ELSER tutorial (#106490)
elastic/elasticsearch: docs/reference/search/search-your-data/semantic-search-elser.asciidoc

[[text-expansion-query]]
==== Semantic search by using the `text_expansion` query

To perform semantic search, use the <<query-dsl-text-expansion-query, `text_expansion` query>>,

> and provide the  query text and the ELSER model ID. The example below uses the

Reviewer: ```suggestion
and provide the query text and the ELSER model ID. The example below uses the
```

[[text-expansion-compound-query]]
==== Combining semantic search with other queries

> You can combine <<query-dsl-text-expansion-query, `text_expansion`>> with other
> queries in a <<compound-queries,compound query>>. For example using a filter clause
> in a <<query-dsl-bool-query>> or a full text query which may or may not use the same

Reviewer: ```suggestion
You can combine <<query-dsl-text-expansion-query, `text_expansion`>> with other
queries in a <<compound-queries,compound query>>. For example, use a filter clause
in a <<query-dsl-bool-query>> or a full text query with the same (or different)
```
